File: indexter-games/src/main/java/com/github/ykrasik/indexter/games/datamodel/persistence/ExcludedPath.java
package com.github.ykrasik.indexter.games.datamodel.persistence;

import com.github.ykrasik.indexter.id.Id;
import lombok.EqualsAndHashCode;
import lombok.NonNull;
import lombok.Value;

import java.nio.file.Path;

/**
 * @author Yevgeny Krasik
 */
@Value
@EqualsAndHashCode(of = "id")
public class ExcludedPath {
    @NonNull private final Id<ExcludedPath> id;
    @NonNull private final Path path;

    @Override
    public String toString() {
        return path.toString();
    }
}

File: gamedex-datamodel/src/main/kotlin/com/github/ykrasik/gamedex/datamodel/ExcludedPath.kt
package com.github.ykrasik.gamedex.datamodel

import com.github.ykrasik.gamedex.common.Id
import java.nio.file.Path

/**
 * User: ykrasik
 * Date: 25/05/2016
 * Time: 12:08
 */
data class ExcludedPath(
    val id: Id<ExcludedPath>,
    val path: Path
)